File: Sep3-SemesterProject/Tier1-Presentation/Domain/DTOs/RegisterDoctorDto.cs
namespace Domain.DTOs;

public class RegisterDoctorDto
{
    public string UserName { get;set; }
    public string Password { get; set; }
    public string FullName { get; set; }
    public string Specialization { get; set; }

    public RegisterDoctorDto(string username, string password, string fullName, string specialization)
    {
        UserName = username;
        Password = password;
        FullName = fullName;
        Specialization = specialization;
    }
}

File: Sep3-SemesterProject/Tier2-BusinessLogic/src/main/java/via/sdj3/proofofconcept_v3/Dto/RegisterDoctorDto.java
package via.sdj3.proofofconcept_v3.Dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class RegisterDoctorDto {
    @JsonProperty("UserName")
    private String UserName;

    @JsonProperty("Password")
    private String Password;

    @JsonProperty("FullName")
    private String FullName;

    @JsonProperty("Specialization")
    private String Specialization;


    public RegisterDoctorDto(String name, String username, String specialization, String password) {
        this.FullName = name;
        this.UserName = username;
        this.Specialization = specialization;
        this.Password = password;
    }

    public RegisterDoctorDto() {
    }

    public String getName() {
        return UserName;
    }

    public void setName(String name) {
        this.UserName = name;
    }

    public String getPassword() {
        return Password;
    }

    public void setPassword(String password) {
        this.Password = password;
    }

    public String getFullName() {
        return FullName;
    }

    public void setFullName(String fullName) {
        FullName = fullName;
    }

    public String getSpecialization() {
        return Specialization;
    }

    public void setSpecialization(String specialization) {
        Specialization = specialization;
    }
}
